src/pages/SettingsPage/InitialPage.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import {
+    TouchableOpacity,
+    View,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import Str from 'expensify-common/lib/str';
+import styles from '../../styles/styles';
+import Text from '../../components/Text';
+import {signOut} from '../../libs/actions/Session';
+import ONYXKEYS from '../../ONYXKEYS';
+import {version} from '../../../package.json';
+import AvatarWithIndicator from '../../components/AvatarWithIndicator';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import {redirect, redirectToLastReport} from '../../libs/actions/App';
+import {
+    Gear, Lock, Profile, Wallet,
+} from '../../components/Icon/Expensicons';
+import HeaderGap from '../../components/HeaderGap';
+import MenuItem from '../../components/MenuItem';
+import ROUTES from '../../ROUTES';
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: PropTypes.shape({
+        // Display name of the current user from their personal details
+        displayName: PropTypes.string,
+
+        // Avatar URL of the current user from their personal details
+        avatar: PropTypes.string,
+    }),
+
+    // Information about the network
+    network: PropTypes.shape({
+        // Is the network currently offline or not
+        isOffline: PropTypes.bool,
+    }),
+
+    // The session of the logged in person
+    session: PropTypes.shape({
+        // Email of the logged in person
+        email: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    myPersonalDetails: {},
+    network: null,
+    session: {},
+};
+
+const menuItems = [
+    {
+        title: 'Profile',
+        icon: Profile,
+        route: 'profile',
+    },
+    {
+        title: 'Preferences',
+        icon: Gear,
+        route: 'preferences',
+    },
+    {
+        title: 'Change Password',
+        icon: Lock,
+        route: 'password',
+    },
+    {
+        title: 'Payments',
+        icon: Wallet,
+        route: 'payments',
+    },
+];
+
+const InitialSettingsPage = ({
+    myPersonalDetails,
+    network,
+    session,
+}) => {
+    // On the very first sign in or after clearing storage these
+    // details will not be present on the first render so we'll just
+    // return nothing for now.
+    if (!myPersonalDetails || _.isEmpty(myPersonalDetails)) {
```

Comment:
most of this page was existing that was copied over from the original SettingsPage. but I agree, not sure if we need both checks.

https://github.com/Expensify/Expensify.cash/pull/1630/files#diff-4a3e2aaac09f347ddfe8aa0c0a88f0a59af21151d683d1cfd8c62e9577663d7eL79-L84

Reply:
Can you confirm and remove it if it's not needed? @marcaaron is this scope creep or is that a good change to include in this PR?

Reply:
updated to just `_.isEmpty(myPersonalDetails)` for now. IMO in general should avoid returning null on pages since it can lead to users stuck being stuck on a white screen. I suggest wrapping the content with a conditional and leaving the header outside to be able to navigate away

Reply:
Seems like `myPersonalDetails` can only be either an empty object or the details. This might just be some left over code in place to protect from them becoming `null` somehow. At some point, Onyx was not removing explicitly `null` values. But we've since fixed that issue. Probably we just didn't think about it very hard.